lib/rules/no-restricted-imports.js

```diff
@@ -262,12 +270,13 @@ module.exports = {
 
             const customMessage = group.customMessage;
             const restrictedImportNames = group.importNames;
+            const restrictedImportNamePattern = !group.importNamePattern ? null : new RegExp(group.importNamePattern, "u");
```

Comment:
```suggestion
            const restrictedImportNamePattern = group.importNamePattern ? new RegExp(group.importNamePattern, "u") : null;
```

Reads better without negations.